src/vs/editor/contrib/inlineEdit/browser/inlineEditController.ts

```diff
@@ -269,10 +272,13 @@ export class InlineEditController extends Disposable {
 		this.editor.pushUndoStop();
 		this.editor.executeEdits('acceptCurrent', [EditOperation.replace(Range.lift(data.range), text)]);
 		if (data.accepted) {
-			this._commandService.executeCommand(data.accepted.id, ...data.accepted.arguments || []);
+			await this._commandService
+				.executeCommand(data.accepted.id, ...(data.accepted.arguments || []))
+				.then(undefined, onUnexpectedExternalError);
 		}
 		this.freeEdit(data);
 		this._currentEdit.set(undefined, undefined);
+		this._isAccepting.set(false, undefined);
```

Comment:
Please use `transaction` here, dont pass in `undefined`. That why, `autorun`s that listen on both observables are only called once.

Reply:
✅